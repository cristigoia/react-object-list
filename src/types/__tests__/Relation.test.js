import React from 'react'
import { shallow } from 'enzyme'
import {
  snapshotTest,
} from 'utils/tests'
import {
  ObjectLink,
  Relation,
} from '../'

describe('Relation type', () => {
  it('snapshot tests', () => {
    const input = {data: null}
    const noAttrs = {data: {id: 1, type: 'TestModel'}}
    const strAttr = {...noAttrs,
      attributes: {
        __str__: 'default',
      },
    }
    const testValues = [
      {},
      {value: input},
      {value: noAttrs},
      {value: strAttr},
      {value: strAttr, tag: 'ref'},
      {value: strAttr, tag: '__str__'},
      {value: strAttr, detail: 'ref'},
      {value: strAttr, detail: '__str__'},
    ]
    testValues.forEach(testValue => {
      snapshotTest(<Relation {...testValue} />)
    })
    const relation = shallow(<Relation value={strAttr} showLink={false} />)
    expect(relation.containsMatchingElement(<ObjectLink value={strAttr} linked={false} />)).toBe(true)
  })
})
