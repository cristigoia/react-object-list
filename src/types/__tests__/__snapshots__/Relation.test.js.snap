// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Relation type snapshot tests 1`] = `
<span
  className="text-muted"
>
  -
</span>
`;

exports[`Relation type snapshot tests 2`] = `
<span
  className="text-muted"
>
  -
</span>
`;

exports[`Relation type snapshot tests 3`] = `
<div
  className=""
>
  <div>
    <div>
      <div>
        TestModel 1
      </div>
    </div>
  </div>
</div>
`;

exports[`Relation type snapshot tests 4`] = `
<div
  className=""
>
  <div>
    <div>
      <div>
        TestModel 1
      </div>
    </div>
  </div>
</div>
`;

exports[`Relation type snapshot tests 5`] = `
<div
  className=""
>
  <div>
    <div>
      <div>
        TestModel 1
      </div>
    </div>
  </div>
</div>
`;

exports[`Relation type snapshot tests 6`] = `
<div
  className=""
>
  <div>
    <div>
      <div>
        TestModel 1
      </div>
    </div>
  </div>
</div>
`;

exports[`Relation type snapshot tests 7`] = `
<div
  className=""
>
  <div>
    <div>
      <div>
        TestModel 1
      </div>
    </div>
  </div>
</div>
`;

exports[`Relation type snapshot tests 8`] = `
<div
  className=""
>
  <div>
    <div>
      <div>
        TestModel 1
      </div>
    </div>
  </div>
</div>
`;
